File: android/src/main/kotlin/com/example/printer_connect/PrinterConnectLogger.kt
package com.example.printer_connect

import android.util.Log
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

// 日志系统：基于 Android Log 封装的分级日志工具。
// 支持 5 个日志级别（ERROR > WARNING > INFO > DEBUG > VERBOSE），
// 通过 setLogLevel 设置输出阈值，仅输出指定级别及以上的日志。
// 每条日志自动添加时间戳前缀，便于调试时定位时序问题。
object PrinterConnectLogger {
    private const val TAG = "PrinterConnect"
    // 当前日志级别阈值，默认为 NONE（不输出任何日志）
    private var currentLogLevel: BleLogLevel = BleLogLevel.NONE
    // 时间戳格式：精确到毫秒，使用 Locale.US 保证格式一致
    private val timeFormatter = SimpleDateFormat("HH:mm:ss.SSS", Locale.US)

    // 设置日志级别阈值，仅输出该级别及以上的日志
    fun setLogLevel(logLevel: BleLogLevel) {
        currentLogLevel = logLevel
    }

    fun logError(message: String) {
        if (!allows(BleLogLevel.ERROR)) return
        Log.e(TAG, withTimestamp(message))
    }

    fun logWarning(message: String) {
        if (!allows(BleLogLevel.WARNING)) return
        Log.w(TAG, withTimestamp(message))
    }

    fun logInfo(message: String) {
        if (!allows(BleLogLevel.INFO)) return
        Log.i(TAG, withTimestamp(message))
    }

    fun logDebug(message: String) {
        if (!allows(BleLogLevel.DEBUG)) return
        Log.d(TAG, withTimestamp(message))
    }

    fun logVerbose(message: String) {
        if (!allows(BleLogLevel.VERBOSE)) return
        Log.v(TAG, withTimestamp(message))
    }

    // 判断指定级别是否允许输出：
    // - NONE 级别始终不输出
    // - 其他级别按 ordinal 值比较，level.ordinal <= currentLogLevel.ordinal 时允许输出
    private fun allows(level: BleLogLevel): Boolean {
        if (currentLogLevel == BleLogLevel.NONE) return false
        return level.ordinal <= currentLogLevel.ordinal
    }

    // 为日志消息添加时间戳前缀：[HH:mm:ss.SSS]
    private fun withTimestamp(message: String): String {
        return "[${timeFormatter.format(Date())}] $message"
    }
}

File: ios/Classes/PrinterConnectLogger.swift
import Foundation

/// 日志系统类（单例模式）。
///
/// 支持 5 个日志级别：error、warning、info、debug、verbose。
/// 通过 setLogLevel 设置输出级别，只有级别值 <= 当前设置的日志才会输出。
/// 日志格式：[时间戳] PrinterConnect:级别 消息内容
final class PrinterConnectLogger {
    /// 共享实例
    static let shared = PrinterConnectLogger()
    private init() {}

    /// 日期格式化器，用于生成日志时间戳
    private lazy var dateFormatter: DateFormatter = {
        let formatter = DateFormatter()
        formatter.dateFormat = "HH:mm:ss.SSS"
        return formatter
    }()

    /// 当前日志级别（默认为 .none 即不输出任何日志）
    private var currentLogLevel: BleLogLevel = .none

    /// 设置日志输出级别
    func setLogLevel(_ logLevel: BleLogLevel) {
        currentLogLevel = logLevel
    }

    /// 输出 ERROR 级别日志
    func logError(_ message: String) {
        guard allows(.error) else { return }
        print("PrinterConnect:ERROR \(withTimestamp(message))")
    }

    /// 输出 WARNING 级别日志
    func logWarning(_ message: String) {
        guard allows(.warning) else { return }
        print("PrinterConnect:WARN \(withTimestamp(message))")
    }

    /// 输出 INFO 级别日志
    func logInfo(_ message: String) {
        guard allows(.info) else { return }
        print("PrinterConnect:INFO \(withTimestamp(message))")
    }

    /// 输出 DEBUG 级别日志
    func logDebug(_ message: String) {
        guard allows(.debug) else { return }
        print("PrinterConnect:DEBUG \(withTimestamp(message))")
    }

    /// 输出 VERBOSE 级别日志
    func logVerbose(_ message: String) {
        guard allows(.verbose) else { return }
        print("PrinterConnect:VERBOSE \(withTimestamp(message))")
    }

    /// 判断指定级别是否应该输出（只有当级别值 <= 当前设置值时才输出）
    private func allows(_ level: BleLogLevel) -> Bool {
        return currentLogLevel != .none && level.rawValue <= currentLogLevel.rawValue
    }

    /// 为日志消息添加时间戳前缀
    private func withTimestamp(_ message: String) -> String {
        let time = dateFormatter.string(from: Date())
        return "[\(time)] \(message)"
    }
}